### ISSUE TITLE
Error in the plugin admin page

### ISSUE THREAD

**Issue by denis-sumin** [VSLCatena/ldap_login#13]
Firstly, thank you for the work on the plugin update for Piwigo 2.9.x. However, I tried to use it and I failed.

I have installed the plugin (Piwigo 2.9.4, PHP 7.0, docker container based on debian), and I see the errors in the admin page:

![image](https://user-images.githubusercontent.com/1160574/46916096-31b8eb00-cfb6-11e8-88cc-990517db8e1b.png)

The configuration is also not saved. Can you hint me what it can come from? My one guess is that you develop on PHP 5.x and that these errors are because of stricter PHP 7. Can it be the case?

**Comment by denis-sumin** [comment]
When I click on Activate in Piwigo admin, there is also this page showing up for a second. Some errors here as well:

![image](https://user-images.githubusercontent.com/1160574/46916153-f8cd4600-cfb6-11e8-8d20-1012d79371f9.png)

**Comment by Kipjr** [comment]
Hi! Thanks for your message. I think your directory is not writeable by your Apache/Nginx user. Furthermore, the errors about the variables are a consequence of the data.dat file that is not yet been made.


Could you confirm this?

________________________________
Van: Denis <notifications@github.com>
Verzonden: zondag 14 oktober 2018 13:42:34
Aan: VSLCatena/ldap_login
CC: Subscribed
Onderwerp: Re: [VSLCatena/ldap_login] Error in the plugin admin page (#13)


When I click on Activate in Piwigo admin, there is also this page showing up for a second. Some errors here as well:

[image]<https://user-images.githubusercontent.com/1160574/46916153-f8cd4600-cfb6-11e8-8d20-1012d79371f9.png>

**Comment by denis-sumin** [comment]
Yep, this was my another guess; I've even tried to open (`chmod 777`) the plugin dir. Could you tell me which directory should I make writeable?
Regarding the `data.dat`, will it be created automatically if I fix the file permissions or should it be created by me?

Thank you!

**Comment by Kipjr** [comment]
The root directory of the plugin should be writable by your www-data user (or whatever user you have to write from web to server)


data.dat will be created if it does not exist:

fopen(filename,'w',include_path,context)

"w" (Write only. Opens and clears the contents of file; or creates a new file if it doesn't exist)

________________________________
Van: Denis <notifications@github.com>
Verzonden: zondag 14 oktober 2018 16:44:42
Aan: VSLCatena/ldap_login
CC: Martie; Comment
Onderwerp: Re: [VSLCatena/ldap_login] Error in the plugin admin page (#13)


Yep, this was my another guess; I've even tried to open (chmod 777) the plugin dir. Could you tell me which directory should I make writeable?
Regarding the data.dat, will it be created automatically if I fix the file permissions or should it be created by me?

Thank you!

**Comment by denis-sumin** [comment]
hmm, then it is strange: the permissions seem to be sufficient

```
root@ce9016fe3631:/var/www/piwigo/plugins# ls -la
drwxrwxrwx.  4 root root 4096 Oct 14 11:01 Ldap_Login
```

```root@ce9016fe3631:/var/www/piwigo/plugins# ls -la Ldap_Login/
total 64
drwxrwxrwx.  4 root root  4096 Oct 14 11:01 .
drwxr-xr-x.  8 root root  4096 Oct 14 11:32 ..
-rwxrwxrwx.  1 root root   570 Oct 14 11:45 .gitignore
-rwxrwxrwx.  1 root root  2406 Oct 14 11:45 Readme.md
drwxrwxrwx.  2 root root  4096 Oct 14 11:01 admin
-rwxrwxrwx.  1 root root   774 Oct 14 11:45 admin.php
-rwxrwxrwx.  1 root root 12245 Oct 14 11:45 class.ldap.php
-rwxrwxrwx.  1 root root  1764 Oct 14 11:45 index.php
drwxrwxrwx. 16 root root  4096 Oct 14 11:01 language
-rwxrwxrwx.  1 root root  5501 Oct 14 11:45 main.inc.php
-rwxrwxrwx.  1 root root  1915 Oct 14 11:45 maintain.inc.php
-rwxrwxrwx.  1 root root    11 Oct 14 11:45 obsolete.list
-rwxrwxrwx.  1 root root   185 Oct 14 11:45 pem_metadata.txt
```

Am I write that the write operation happens in line [80](https://github.com/VSLCatena/ldap_login/blob/master/class.ldap.php#L80) of `class.ldap.php`? Then it seems that the constant (`LDAP_LOGIN_PATH`) error which I show in the second screen shot explains the issue, no?

**Comment by Kipjr** [comment]
try "chown www-data:www-data /var/www/piwigo/plugins/Ldap_Login "

________________________________
Van: Denis <notifications@github.com>
Verzonden: zondag 14 oktober 2018 18:01:48
Aan: VSLCatena/ldap_login
CC: Martie; Comment
Onderwerp: Re: [VSLCatena/ldap_login] Error in the plugin admin page (#13)


hmm, then it is strange: the permissions seem to be sufficient

root@ce9016fe3631:/var/www/piwigo/plugins# ls -la
drwxrwxrwx.  4 root root 4096 Oct 14 11:01 Ldap_Login


total 64
drwxrwxrwx.  4 root root  4096 Oct 14 11:01 .
drwxr-xr-x.  8 root root  4096 Oct 14 11:32 ..
-rwxrwxrwx.  1 root root   570 Oct 14 11:45 .gitignore
-rwxrwxrwx.  1 root root  2406 Oct 14 11:45 Readme.md
drwxrwxrwx.  2 root root  4096 Oct 14 11:01 admin
-rwxrwxrwx.  1 root root   774 Oct 14 11:45 admin.php
-rwxrwxrwx.  1 root root 12245 Oct 14 11:45 class.ldap.php
-rwxrwxrwx.  1 root root  1764 Oct 14 11:45 index.php
drwxrwxrwx. 16 root root  4096 Oct 14 11:01 language
-rwxrwxrwx.  1 root root  5501 Oct 14 11:45 main.inc.php
-rwxrwxrwx.  1 root root  1915 Oct 14 11:45 maintain.inc.php
-rwxrwxrwx.  1 root root    11 Oct 14 11:45 obsolete.list
-rwxrwxrwx.  1 root root   185 Oct 14 11:45 pem_metadata.txt


Am I write that the write operation happens in line 80<https://github.com/VSLCatena/ldap_login/blob/master/class.ldap.php#L80> of class.ldap.php? Then it seems that the constant (LDAP_LOGIN_PATH) error which I show in the second screen shot explains the issue, no?

**Comment by denis-sumin** [comment]
I've submitted the fix to this issue: #14
